app/services/EstimatedTaxLiabilityService.scala

```diff
@@ -18,25 +18,31 @@ package services
 
 import javax.inject.{Inject, Singleton}
 
-import connectors.EstimatedTaxLiabilityConnector
+import connectors.LastTaxCalculationConnector
 import models._
 import play.api.Logger
+import play.api.http.Status
 import uk.gov.hmrc.play.http.HeaderCarrier
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
-class EstimatedTaxLiabilityService @Inject()(val estimatedTaxLiabilityConnector: EstimatedTaxLiabilityConnector) {
+class EstimatedTaxLiabilityService @Inject()(val lastTaxCalculationConnector: LastTaxCalculationConnector) {
 
-  def getEstimatedTaxLiability(mtditid: String)(implicit headerCarrier: HeaderCarrier): Future[EstimatedTaxLiabilityResponseModel] = {
-    Logger.debug("[EstimatedTaxLiabilityService][getEstimateTaxLiability] - Requesting Estimate Liability from Backend via Connector")
-    estimatedTaxLiabilityConnector.getEstimatedTaxLiability(mtditid).map {
-      case success: EstimatedTaxLiability =>
-        Logger.debug(s"[EstimatedTaxLiabilityService][getEstimateTaxLiability] - Retrieved Estimated Tax Liability: \n\n$success")
-        success
-      case error: EstimatedTaxLiabilityError =>
-        Logger.debug(s"[EstimatedTaxLiabilityService][getEstimateTaxLiability] - Error Response Status: ${error.status}, Message: ${error.message}")
+  def getLastEstimatedTaxCalculation(nino: String)(implicit headerCarrier: HeaderCarrier): Future[LastTaxCalculationResponseModel] = {
+    Logger.debug("[EstimatedTaxLiabilityService][getLastEstimatedTaxCalculation] - Requesting Last Tax from Backend via Connector")
+    lastTaxCalculationConnector.getLastEstimatedTax(nino).map {
+      case success: LastTaxCalculation =>
+        Logger.debug(s"[EstimatedTaxLiabilityService][getLastEstimatedTaxCalculation] - Retrieved Estimated Tax Liability: \n\n$success")
+        success match {
+          case LastTaxCalculation(_,_,Some(amount)) => success
+          case LastTaxCalculation(_,_,None) =>
```

Comment:
This is an interesting one. The call could legitimately be successful but return no last calculation. I.e. if the user has not submitted any data yet then we won't be able to show a calculation.

I think we need to raise this with Oli, Rohima and Laura to see what there views are? As maybe we should handle this as a valid scenario and display different content.